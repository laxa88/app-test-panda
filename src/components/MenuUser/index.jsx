import React from 'react';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const MenuUser = () => (
  <div>
    <FontAwesomeIcon icon={['fas', 'user']} />
    John | Log out
  </div>
);

export default MenuUser;
