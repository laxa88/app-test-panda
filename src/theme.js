import React from 'react';

export const panda = {
  primaryColor: '#33A00E',
};

export const sakura = {
  primaryColor: 'pink',
};

export const ThemeContext = React.createContext({
  theme: panda,
});
