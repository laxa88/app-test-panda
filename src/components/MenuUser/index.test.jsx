import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { currentTheme } from '../../theme';
import MenuUser, { StyledContainer } from './index';

describe('<MenuUser />', () => {
  it('renders', () => {
    const result = renderer.create(<MenuUser />);

    expect(result.root.children[0].props.children[0].props.icon).toEqual(['fas', 'user']);
    expect(result.root.children[0].props.children[1].props.children).toBe('John | Log out');
  });

  it('renders StyledContainer', () => {
    const result = renderer.create(<StyledContainer theme={currentTheme} />);

    expect(result.toJSON()).toHaveStyleRule(
      'color',
      currentTheme.primaryColor1,
      { modifier: '& svg' },
    );
  });
});
