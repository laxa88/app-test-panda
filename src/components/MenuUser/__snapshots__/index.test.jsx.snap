// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuUser /> renders 1`] = `
<div>
  Mocked_FontAwesomeIcon
  John | Log out
</div>
`;
